### ISSUE TITLE
Upgrade guide of `0.20` for users used to use config `base64::URL_SAFE_NO_PAD`

### ISSUE THREAD

**Issue by Xuanwo** [marshallpierce/rust-base64#206]
Users who used to use `base64::decode_config(&secret_key, base64::URL_SAFE_NO_PAD)` will be broken after upgrading from `0.13` to `0.20`. This issue intends to provide a quick upgrade guide.

Implement a engine:

```rust
/// The url safe engine for base64.
pub const BASE64_URL_SAFE_ENGINE: base64::engine::fast_portable::FastPortable =
    base64::engine::fast_portable::FastPortable::from(
        &base64::alphabet::URL_SAFE,
        base64::engine::fast_portable::NO_PAD,
    );
```

And replace all usage:

```diff
- base64::encode_config(&code_verifier_bytes, base64::URL_SAFE_NO_PAD)
+ base64::encode_engine(&code_verifier_bytes, &BASE64_URL_SAFE_ENGINE)
- base64::encode_config(&hasher.finalize(), base64::URL_SAFE_NO_PAD)
+ base64::encode_engine(&hasher.finalize(), &BASE64_URL_SAFE_ENGINE)
```

More info could be found at https://github.com/mozilla/sccache/pull/1477.

**Comment by Xuanwo** [comment]
`RELEASE-NOTES` doesn't have enough information for this upgrade. How about adding this issue's content into `RELEASE-NOTES`? I'm willing to submit a PR for this.

Cc @marshallpierce

**Comment by marshallpierce** [comment]
Thanks for the report. Based on feedback in https://github.com/marshallpierce/rust-base64/issues/205 I'll be quickly following 0.20 with 0.21, and part of that will be better documentation. I'll address this as part of that.

**Comment by djmitche** [comment]
It would be great to have a complete migration guide, particularly since some of the behavior changes are a bit subtle.  Base64 strings containing url-unsafe characters are pretty unusual, and probably easy to miss in the unit tests of crates using this one.  And, a shift in expectations for padding on decoding might similarly be missed, if the test values are always correctly padded.

My understanding, confirmed with some experimentation:

| 0.13.1 Config   | 0.20.0 alphabet | `encode_padding` | `decode_padding_mode` |
|-----------------|-----------------|------------------|-----------------------|
| STANDARD        | STANDARD        | true             | Indifferent           |
| STANDARD_NO_PAD | STANDARD        | false            | Indifferent           |
| URL_SAFE        | URL_SAFE        | true             | Indifferent           |
| URL_SAFE_NO_PAD | URL_SAFE        | false            | Indifferent           |

I think the surprising thing to me was the last column -- especially with `.._NO_PAD`, I assumed `DecodePaddingMode::RequireNone`.  Since my examples didn't have padding, I silently introduced a bug.

**Comment by marshallpierce** [comment]
There wasn't the ability to express `RequireNone` before 0.20. I don't want to surprise users, but it seemed the most in keeping with the intent of the NO_PAD configs, and in general I want it to be an explicit user choice to be _less_ restrictive (which is why the with-padding configs now _require_ it). Anyway, in 0.21 I'll have more in the release notes on migration, which should be out in a few days.

**Comment by marshallpierce** [comment]
Take a look at https://github.com/marshallpierce/rust-base64/pull/207. I've released that as `0.21.0-beta.1` also to make it easy to try out.

**Comment by Alexey-N-Chernyshov** [comment]
There is a 0.21.0 release with [release notes.](https://github.com/marshallpierce/rust-base64/blob/master/RELEASE-NOTES.md#migration). Seems that issue is resolved.
